cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -153,7 +149,7 @@ protected void run() {
         // If nothing has been processed (e.g. empty fetch from Kafka, fail to append anything to appender),
         // Sleep until the earliest event in the buffer is time to be written out.
         if (!hasMessageProcessed) {
-          long sleepMillis = config.getBufferMillis();
+          long sleepMillis = config.getEventDelayMillis();
           if (!eventQueue.isEmpty()) {
```

Comment:
I don't get your comment. The whole computation of the sleep time logic is described in the commit couple lines above.